projects/15_basic_material/main.cpp

```diff
@@ -642,6 +765,51 @@ void ProjApp::Setup()
     // Samplers
     SetupSamplers();
 
+    // Env draw data
+    {
+        grfx::DescriptorSetLayoutCreateInfo createInfo = {};
+        createInfo.bindings.push_back({grfx::DescriptorBinding{0, grfx::DESCRIPTOR_TYPE_UNIFORM_BUFFER, 1, grfx::SHADER_STAGE_ALL_GRAPHICS}});
+        createInfo.bindings.push_back({grfx::DescriptorBinding{1, grfx::DESCRIPTOR_TYPE_SAMPLED_IMAGE, 1, grfx::SHADER_STAGE_ALL_GRAPHICS}});
+        createInfo.bindings.push_back({grfx::DescriptorBinding{2, grfx::DESCRIPTOR_TYPE_SAMPLER, 1, grfx::SHADER_STAGE_ALL_GRAPHICS}});
+        PPX_CHECKED_CALL(GetDevice()->CreateDescriptorSetLayout(&createInfo, &mEnvDrawLayout));
+
+        PPX_CHECKED_CALL(GetDevice()->AllocateDescriptorSet(mDescriptorPool, mEnvDrawLayout, &mEnvDrawSet));
+
+        // Scene constants
+        grfx::BufferCreateInfo bufferCreateInfo      = {};
+        bufferCreateInfo.size                        = PPX_MINIMUM_CONSTANT_BUFFER_SIZE;
+        bufferCreateInfo.usageFlags.bits.transferSrc = true;
+        bufferCreateInfo.memoryUsage                 = grfx::MEMORY_USAGE_CPU_TO_GPU;
+        PPX_CHECKED_CALL(GetDevice()->CreateBuffer(&bufferCreateInfo, &mCpuEnvDrawConstants));
+
+        bufferCreateInfo.usageFlags.bits.transferDst   = true;
+        bufferCreateInfo.usageFlags.bits.uniformBuffer = true;
+        bufferCreateInfo.memoryUsage                   = grfx::MEMORY_USAGE_GPU_ONLY;
+        PPX_CHECKED_CALL(GetDevice()->CreateBuffer(&bufferCreateInfo, &mGpuEnvDrawConstants));
+
+        grfx::WriteDescriptor write = {};
+        write.binding               = 0;
+        write.arrayIndex            = 0;
+        write.type                  = grfx::DESCRIPTOR_TYPE_UNIFORM_BUFFER;
+        write.bufferOffset          = 0;
+        write.bufferRange           = PPX_WHOLE_SIZE;
+        write.pBuffer               = mGpuEnvDrawConstants;
+        PPX_CHECKED_CALL(mEnvDrawSet->UpdateDescriptors(1, &write));
```

Comment:
I am curious, is there a reason we can't/shouldn't use one UpdateDescriptors call for all 3 writes?